misc/systemd/kubo-hardened.service

```diff
@@ -51,26 +51,26 @@ CapabilityBoundingSet=CAP_NET_BIND_SERVICE
 # enable for 1-1024 port listening
 #AmbientCapabilities=CAP_NET_BIND_SERVICE 
 # enable to specify a custom path see docs/environment-variables.md for further documentations
-#Environment=IPFS_PATH=/custom/ipfs/path
+#Environment=IPFS_PATH=/custom/kubo/path
 # enable to specify a higher limit for open files/connections
 #LimitNOFILE=1000000
 
 #don't use swap
 MemorySwapMax=0
 
-# Don't timeout on startup. Opening the IPFS repo can take a long time in some cases (e.g., when
-# badger is recovering) and migrations can delay startup.
+# Don't timeout on startup. Opening the Kubo repo can take a long time in some cases (e.g., when
+# badger is recovering).
 #
 # Ideally, we'd be a bit smarter about this but there's no good way to do that without hooking
-# systemd dependencies deeper into go-ipfs.
+# systemd dependencies deeper into Kubo.
 TimeoutStartSec=infinity
 
 Type=notify
-User=ipfs
-Group=ipfs
-StateDirectory=ipfs
+User=kubo
+Group=kubo
+StateDirectory=kubo
 Environment=IPFS_PATH="${HOME}"
-ExecStart=/usr/bin/ipfs daemon --init --migrate
+ExecStart=/usr/bin/kubo daemon --enable-gc
```

Comment:
Why did you remove `--init` `--migrate` and added `--enable-gc`? iiuc:
- Suddenly enabling GC for existing users may cause data loss
- Disabling automatic init and migraitons will be a regression in current UX

Reply:
Well, regarding `--init` is usually handled by install scripts, like mine for the ArchLinux dev package:

```shellscript
# in case the daemon was never started, generate a configuration

if [ ! -f /var/lib/ipfs/config ]; then
  systemctl restart systemd-sysusers.service
  mkdir -p /var/lib/ipfs/
  chown ipfs: -R /var/lib/ipfs/
  su ipfs -s /bin/bash -c "export IPFS_PATH='/var/lib/ipfs/'; ipfs init"
fi
```

The migration can since 0.14 supports migration without starting the daemon, do as well. The only reason there was a "--migrate" was - that this wasn't implemented.

"--enable-gc" not to set was an oversight by me, as the default config sets a maximum size for the repo, it would break the principle of least surprise if the daemon would not clean up.

The garbage collector cannot cause data loss by design – since only unpinned elements will be deleted.